brvs-core/src/main/java/iroha/validation/rules/RuleMonitor.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright D3 Ledger, Inc. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package iroha.validation.rules;
+
+import static com.d3.commons.util.ThreadUtilKt.createPrettySingleThreadPool;
+
+import io.reactivex.Scheduler;
+import io.reactivex.schedulers.Schedulers;
+import iroha.protocol.Commands.Command;
+import iroha.protocol.Commands.SetAccountDetail;
+import iroha.protocol.TransactionOuterClass.Transaction.Payload.ReducedPayload;
+import iroha.validation.listener.BrvsIrohaChainListener;
+import iroha.validation.utils.ValidationUtils;
+import iroha.validation.validators.Validator;
+import java.util.Objects;
+import java.util.Set;
+import jp.co.soramitsu.iroha.java.QueryAPI;
+import jp.co.soramitsu.iroha.java.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+
+/**
+ * Class for listening for blocks and react on rules configuration updates
+ */
+public class RuleMonitor {
+
+  private static final Logger logger = LoggerFactory.getLogger(RuleMonitor.class);
+
+  private final Scheduler scheduler = Schedulers.from(createPrettySingleThreadPool(
+      "rule-monitor", "chain-listener"
+  ));
+  private final QueryAPI queryAPI;
+  private final BrvsIrohaChainListener irohaChainListener;
+  private final String repositoryAccountId;
+  private final String settingsAccountId;
+  private final String setterAccountId;
+  private final Validator validator;
+  private volatile boolean isStarted = false;
```

Comment:
No need to make if false. It's false by default

Reply:
No need to make it volatile, because you read and modify this field in `synchronized` block